utk_shiftlog/server.lua

```diff
@@ -83,15 +113,22 @@ AddEventHandler("utk_sl:jobchanged", function(old, new, method)
             end
         end
     end
-    if new == "police" or new == "ambulance" then
-        table.insert(timers[new], {id = xPlayer.source, identifier = xPlayer.identifier, name = xPlayer.name, time = os.time(), date = os.date("%d/%m/%Y %X")})
+    if new == "police" or new == "ambulance" or new == "mechanic" or new == "cardealer" then
+        table.insert(timers[new], {id = xPlayer.source, identifier = xPlayer.identifier, firstname = firstname, lastname = lastname, time = os.time(), date = os.date("%d/%m/%Y %X")})
     end
+
 end)
 
 AddEventHandler("playerDropped", function(reason)
     local id = source
     local header = nil
     local color = nil
+    local xPlayer = ESX.GetPlayerFromId(source)

```

Comment:
this won't work always since ESX object of that player is deleted in es_extended, you need to get identifier with cfx native
local identifier = GetPlayerIdentifiers(id)[1] -- 1 is steam